Parking_Zone.Test/Controllers/ParkingZoneControllerTest.cs

```diff
@@ -0,0 +1,309 @@
+        Assert.Equivalent(redirectResult.ActionName, "Index");
+        Assert.True(_parkingZoneController.ModelState.IsValid);
+        _parkingZoneServiceMoq.Verify(x => x.Create(It.IsAny<ParkingZone>()), Times.Once());
+    }
+
+    #endregion
+
+    #region Edit
+
+    [Fact]
+    public void GivenId_WhenEditIsCalled_ThenReturnsNotFoundResult()
+    {
+        //Arrange
+        _parkingZoneServiceMoq.Setup(x => x.GetById(id));
+
+        //Act
+        var result = _parkingZoneController.Edit(id);
+
+        //Assert
+        Assert.IsType<NotFoundResult>(result);
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Fact]
+    public void GivenId_WhenEditIsCalled_ThenReturnsViewResultWithVM()
+    {
+        //Arrange
+        EditVM expectedEditVM = new(_parkingZone);
+
+        _parkingZoneServiceMoq
+            .Setup(x => x.GetById(id))
+            .Returns(_parkingZone);
+
+        //Act
+        var viewResult = _parkingZoneController.Edit(id);
+
+        //Assert
+        var model = Assert.IsType<ViewResult>(viewResult).Model;
+        Assert.IsAssignableFrom<EditVM>(model);
+        Assert.Equal(JsonSerializer.Serialize(expectedEditVM), JsonSerializer.Serialize(model));
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Fact]
+    public void GivenIdAndEditVM_WhenPostEditIsCalled_ThenReturnsNotFoundResult()
+    {
+        //Arrange
+        EditVM editVM = new();
+        ParkingZone existingParkingZone = new() { Id = 2 };
+
+        _parkingZoneServiceMoq
+            .Setup(x => x.GetById(id))
+            .Returns(existingParkingZone);
+
+        //Act
+        var result = _parkingZoneController.Edit(id, editVM);
+
+        //Assert
+        Assert.IsType<NotFoundResult>(result);
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Theory]
+    [InlineData("Name", "Required")]
+    [InlineData("Address", "Required")]
+    public void GivenIdAndEditVM_WhenPostEditIsCalled_ThenModelStateIsFalseAndReturnsViewResult(string key, string errorMessage)
+    {
+        //Arrange
+        EditVM editVM = new(_parkingZone);
+        EditVM expectedEditVM = new(_parkingZone);
+
+        _parkingZoneController.ModelState.AddModelError(key, errorMessage);
+        _parkingZoneServiceMoq
+            .Setup(x => x.GetById(id))
+            .Returns(_parkingZone);
+
+        //Act
+        var result = _parkingZoneController.Edit(id, editVM);
+
+        //Assert
+        var model = Assert.IsType<ViewResult>(result).Model;
+        Assert.IsAssignableFrom<EditVM>(model);
+        Assert.False(_parkingZoneController.ModelState.IsValid);
+        Assert.Equal(JsonSerializer.Serialize(expectedEditVM), JsonSerializer.Serialize(model));
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Fact]
+    public void GivenIdAndEditVM_WhenEditIsCalled_ThenModelStateIsTrueAndReturnsRedirectToActionResult()
```

Comment:
For case model state is true, you dont need to create separate test case. Just add an additional parameter for the test above called "bool expectedResult". With this way you can check both for false and true in one test